doc/source/programs/gdal_raster_pixelinfo.rst

```diff
@@ -0,0 +1,238 @@
+.. _gdal_raster_pixelinfo:
+
+================================================================================
+``gdal raster pixelinfo``
+================================================================================
+
+.. versionadded:: 3.11
+
+.. only:: html
+
+    Return information on a pixel of a raster dataset
+
+.. Index:: gdal raster pixelinfo
+
+Synopsis
+--------
+
+.. program-output:: gdal raster pixelinfo --help-doc
+
+Description
+-----------
+
+:program:`gdal raster pixelinfo` provide a mechanism to query information about
+a pixel given its location in one of a variety of coordinate systems.
+
+It supports outputting either as GeoJSON or CSV.
+
+The following items will be reported (when known):
+
+- Input coordinates
+- Input coordinates converted to column, line
+- Pixel value per selected band(s), with unscaled value when offset/scale is associated to the band
+- For VRT files, which file(s) contribute to the pixel value.
+
+The following options are available:
+
+Standard options
+++++++++++++++++
+
+.. option:: -f, --of, --format, --output-format geojson|csv
+
+    Which output format to use. Default is GeoJSON.
+
+.. include:: gdal_options/oo.rst
+
+.. include:: gdal_options/if.rst
+
+.. option:: -b <band>
+
+    Selects a band to query. Multiple bands can be listed. By default all
+    bands are queried.
+
+.. option:: -p, --pos, --position <column,line> or <X,Y>
+
+    Required.
+    This can be specified either as an option, a positional value after the
+    dataset name, or when called from :program:`gdal`, as (space separated)
+    values provided on the standard input.
+
+    By default, when :option:`--position-type` is not specified, or set to ``pixel``,
+    this is a column, line tuple (possibly with fractional part).
+    If :option:`--position-type` is set to ``geoloc``, this is a georeferenced
+    coordinate expressed in the CRS of the dataset.
+    If :option:`--position-crs` is specified, this is a georeferenced
+    coordinate expressed in the CRS defined by :option:`--position-crs`.
+
+    X means always longitude or easting, Y means always latitude or northing.
+
+    Several x,y tuples may be specified.
```

Comment:
Thoughts on the ability to read positions from a dataset?

Reply:
I would see that as a potential follow-up enhancement